skimage/measure/block.py

```diff
@@ -13,11 +15,15 @@ def block_reduce(image, block_size, func=np.sum, cval=0):
         Array containing down-sampling integer factor along each axis.
     func : callable
         Function object which is used to calculate the return value for each
-        local block. This function must implement an ``axis`` parameter such
-        as ``numpy.sum`` or ``numpy.min``.
+        local block. This function must implement an ``axis`` parameter.
+        Primary functions are ``numpy.sum``, ``numpy.min``, ``numpy.max``,
+        ``numpy.mean`` and ``numpy.median``.
     cval : float
         Constant padding value if image is not perfectly divisible by the
         block size.
+    **func_kwargs :
+        Flag arguments passed to func. Noteably useful for passing dtype
+        argument to ``np.mean``.
```

Comment:
I'm not positive that we want to unpack these in the function signature by default, as opposed to providing a dict. For example, (side note: looks like cval isn't used at all, which is a problem! But not *your* problem, @justinkterry!), if we were to add a `mode=` argument for the padding, and the user wanted to pass a function for that takes a `mode=` keyword argument also, we now have a namespace conflict that is difficult to resolve.

I'm not positive about this, but my instinct is to make func_kwargs a single dictionary input. What do others think?